src/Minigames/Number_Guesser_UI.py

```diff
@@ -0,0 +1,67 @@
+from quart import Blueprint
+import asyncio
+from socketio import AsyncServer
+from Minigames.Minigame import Minigame
+from .Number_Guesser import Number_Guesser
+
+class Number_Guesser_UI(Minigame):
+    def __init__(self, sio: AsyncServer, blueprint: Blueprint, name=__name__):
+        super().__init__(sio, blueprint, name)
+        self._game = None
+
+        @self._sio.on('join_game')
+        async def on_join_game(sid: str, data):
+            player_id = data['player_id']
+            if len(self.get_players()) < 2 and player_id not in self.get_players():
+                self._players.append(player_id)
+                await self._sio.enter_room(sid, self.get_name())
+                await self._sio.emit('joined', {'player_id': player_id}, room=sid)
+
+        @self._sio.on('set_number')
+        async def on_set_number(sid: str, data):
+            player_id = data['player_id']
+            if player_id == self._players[0]:  # Only the first player sets the number
+                number = data.get('number')
+                try:
+                    self._game.set_number(number)
+                    await self._sio.emit('number_set', {'status': 'success'}, room=self.get_name())
+                except ValueError as e:
+                    await self._sio.emit('number_set', {'status': 'error', 'message': str(e)}, room=sid)
+
+        @self._sio.on('guess_number')
+        async def on_guess_number(sid: str, data):
+            player_id = data['player_id']
+            if player_id == self._players[1]:  # Only the second player guesses
+                guess = data.get('guess')
+                try:
+                    result = self._game.guess_number(guess)
+                    await self._sio.emit('guess_result', {'result': result}, room=self.get_name())
+                    if result['status'] == 'won' or result['status'] == 'lost':
+                        self.cancel()
```

Comment:
self.cancel() should only be called, when the game is actually cancelled. Here the game is finished , therefore the winner should be returned. (Not as a literal return statement here)

Reply:
The statuses used here do not match the ones used in Number_Guesser.